spdx/parsers/tagvaluebuilders.py

```diff
@@ -1031,28 +1031,39 @@ def __init__(self):
         # FIXME: this state does not make sense
         self.reset_file_stat()
 
+    def _build_file_valid(self, doc):
+        """
+        Return true if building a File for referenced document is valid.
+        """
+        if self.has_package(doc) and self.has_file(doc):
+            return True
+        if not self.package_set and self.has_unpackaged_file(doc):
+            return True
+        return False
+
     def set_file_name(self, doc, name):
         """
-        Raise OrderError if no package defined.
+        Set the file name.
         """
         if self.has_package(doc):
             doc.packages[-1].files.append(file.File(name))
-            # A file name marks the start of a new file instance.
-            # The builder must be reset
-            # FIXME: this state does not make sense
-            self.reset_file_stat()
-            return True
         else:
-            raise OrderError("File::Name")
+            # Starting with SPDX 2.0, file entries may proceed any package information section
+            doc.unpackaged_files.append(file.File(name))
+        # A file name marks the start of a new file instance.
+        # The builder must be reset
+        # FIXME: this state does not make sense
+        self.reset_file_stat()
+        return True
 
     def set_file_spdx_id(self, doc, spdx_id):
         """
         Set the file SPDX Identifier.
-        Raise OrderError if no package or no file defined.
+        Raise OrderError if no package/unpackaged file is defined.
```

Comment:
This reads a little bit ambiguously w.r.t. the package itself. Maybe it would be a bit more clear as:
```
Raise OrderError if no package, package file, or unpackaged file is defined.
```